### PR TITLE
feat: Opt out from cache

### PR DESCRIPTION
**Pull request by ptbrowne**
Ability to opt out from cache, by setting a flag on the client.
The flag then flows to the HTTP server through a header. The server decides then whether to use a cache.

- feat: Pass cache from graphql context
- feat: Separate DebugPanel from GqlDebug
- refactor: Extract graphql operations controller out of gql debug
- feat: Better flag, gains observability
- Add devtool for flags

Fixes: https://github.com/visualize-admin/visualization-tool/issues/971

### CODE REVIEW

**Comment by Rdataflow** [comment]
@ptbrowne looks great. just a tiny thing: could you drop the hostname `http://localhost:3000/api/graphql?...` and just put `/api/graphql?...` in the links in the flamegraph? this would help to omit manual regex 😄 
```
http://localhost:3000/api/graphql?query=query%20Themes(%24sourceType%3A%20String!%2C%20%24sourceUrl%3A%20String!%2C%20%24locale%3A%20String!)%20%7B%0A%20%20themes(sourceType%3A%20%24sourceType%2C%20sourceUrl%3A%20%24sourceUrl%2C%20locale%3A%20%24locale)%20%7B%0A%20%20%20%20iri%0A%20%20%20%20label%0A%20%20%20%20__typename%0A%20%20%7D%0A%7D%0A

**Comment by ptbrowne** [comment]
@Rdataflow Done 👍

File: app/flags/flag.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+const flag = function flag(...args: [FlagName] | [FlagName, FlagValue]) {
+  if (args.length === 1) {
+    return store.get(args[0]);
+  } else {
+    const [name, value] = args;
+    console.log("setting", name, value);
+    store.set(name, value);
+    return value;
+  }
+};
```
Comment (bprusinowski) [inline app/flags/flag.tsx:22]: 👌

File: app/flags/useFlag.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { useState, useEffect } from "react";
+
+import { flag } from "./flag";
+
+export default function useFlag(name: string) {
+  const [flagValue, setFlag] = useState(() => flag(name));
+  useEffect(() => {
+    const handleChange = (changed: string) => {
+      if (changed === name) {
+        console.log("ahndle change", changed);
```
Comment (bprusinowski) [inline app/flags/useFlag.ts:10]: `handle*`

File: app/gql-flamegraph/devtool.tsx [hunk 3]
```diff
@@ -460,10 +464,46 @@ const DebugPanel = () => {
...
+        return (
+          <Box key={f} sx={{ display: "flex", gap: "1rem" }}>
+            <Typography variant="body2">{f}</Typography>
+            <FlagSwitch flagName={f} />
+          </Box>
+        );
+      })}
+    </>
+  );
+};
```
Comment (bprusinowski) [inline app/gql-flamegraph/devtool.tsx:490]: 💯

**Review by bprusinowski — APPROVED** [review]
LGTM, super nice with flags! 🎏

**Pull request merged**